src/Symfony/Bundle/FrameworkBundle/Test/HttpClientAssertionsTrait.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Component\HttpClient\DataCollector\HttpClientDataCollector;
+
+/*
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ */
+
+trait HttpClientAssertionsTrait
+{
+    public static function assertHttpClientHasBeenCalledForUrl(KernelBrowser $client, string $httpClientId, string $expectedUrl, string $expectedMethod = 'GET', string|array $expectedBody = null): void
+    {
+        if (!($profile = $client->getProfile())) {
+            static::fail('The Profiler must be enabled for the current request. Please ensure you have called $client->enableProfiler() before making the request.');
+        }
+
+        /** @var HttpClientDataCollector $httpClientDataCollector */
+        $httpClientDataCollector = $profile->getCollector('http_client');
+        $expectedUrlHasBeenFound = false;
+
+        foreach ($httpClientDataCollector->getClients()[$httpClientId]['traces'] as $trace) {
```

Comment:
we should check that `$httpClientDataCollector->getClients()[$httpClientId]` exists first